draft-ietf-oauth-device-flow.xml

```diff
@@ -302,69 +323,96 @@
    Cache-Control: no-store
 
    {
-     "device_code":"74tq5miHKB",
-     "user_code":"94248",
-     "verification_uri":"http://www.example.com/device",
-     "interval"=5
+     "device_code":"GMMhmHCXhWEzkobqIHGG_EnNYYsAkukHspeYUk9E8",
+     "user_code":"WDJB-MJHT",
+     "verification_uri":"https://www.example.com/device",
+     "expires_in" : 1800,
+     "interval": 5
    }
 ]]></artwork></figure>
 	</t> 
 
 </section> 
 
-<section title="User Instruction">
+<section title="User Instruction" anchor="UserInstruction">
 
   <t>
-    After receiving a successful Authorization Response, 
-    the client displays the end-user code and the end-user verification
-    URI to the end-user, and instructs the end-user to visit the URI
-    using a user-agent and enter the end-user code.
+    After receiving a successful Authorization Response, the client displays or
+    otherwise communicates the <spanx style="verb">user_code</spanx> and the
+    <spanx style="verb">verification_uri</spanx> to the end-user, and instructs
+    the them to visit the URI in a user agent on a secondary device
+    (for example, in browser on their mobile phone), and enter the user code.
   </t>
 	
 	<t>
-	  The end-user manually types the provided verification URI and
-	  authenticates with the authorization server.  The authorization
-	  server prompts the end-user to authorize the client's request by
-	  entering the end-user code provided by the client.  Once the end-user
-	  approves or denies the request, the authorization server informs the
-	  end-user to return to the device for further instructions.
-	</t> 
+	  The end-user navigates to the <spanx style="verb">verification_uri</spanx>
+    and authenticates with the authorization server.  The authorization server
+    prompts the end-user to identify the device authorization session by
+    entering the <spanx style="verb">user_code</spanx> provided by the client.
+    The authorization server SHOULD then inform the user about the action they
```

Comment:
Two comments: 

1) I don't think that this is an interoperability related SHOULD. 
2) Why is this a "should" rather than a "must"?  I would instead make it a non-RFC 2119 "must". 

s/The authorization server SHOULD then inform the user.../The authorization server must inform the user ..."